tools/src/launcher/Main.hs

```diff
@@ -100,6 +100,20 @@ type M a = (HasConfigurations, HasCompileInfo) => Log.LoggerNameBox IO a
 
 data Executable = EWallet | ENode | EUpdater
 
+-- Node, its args, node log
+data NodeData = NodeData
+    { _nodePath    :: FilePath
+    , _nodeArgs    :: [Text]
+    , _nodeLogPath :: Maybe FilePath }
+
+-- Updater Node, its args, node log, archive path
+data UpdaterData = UpdaterData
+    { _uNodePath          :: FilePath
+    , _uNodeArgs          :: [Text]
+    , _uNodeLogPath       :: Maybe FilePath
```

Comment:
Are you sure it's the correct name? AFAIU, the third field is the path to windows runner.